### PR TITLE
Upgrade Azure.Idenntity to 1.12.0 For CVE

### PR DESCRIPTION
**Pull request by msJinLei**
## Mandatory Checklist

- Please choose the target release of Azure PowerShell. (⚠️**Target release** is a different concept from **API readiness**. Please click below links for details.)
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] No need for a release

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Review by BethanyZhou — DISMISSED** [review]
Looks good to me

File: src/Accounts/Authentication/Authentication.csproj [hunk 1]
```diff
@@ -12,11 +12,11 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Azure.Identity" Version="1.11.2" />
+    <PackageReference Include="Azure.Identity" Version="1.12.0-beta.3" />
```
Comment (NoriZC) [inline src/Accounts/Authentication/Authentication.csproj:15]: Is the `beta` version stable for us to consume? As Identity looks like a key dependency for us. We might prefer a stable version.

File: src/Synapse/Synapse/Common/AzureSessionCredential.cs [hunk 2]
```diff
@@ -12,16 +12,16 @@
 // limitations under the License.
```
Comment (NoriZC) [inline src/Synapse/Synapse/Common/AzureSessionCredential.cs:12]: Do the affected modules needed to have a release with the change? If yes, could you please update the changelog for it?

File: src/Accounts/Authentication/Authentication.csproj (hunk 1 above)
Comment (isra-fel) [reply src/Accounts/Authentication/Authentication.csproj:15]: https://www.nuget.org/packages/Azure.Identity/1.12.0
We should be using the GA version

File: src/Accounts/AssemblyLoading/ConditionalAssemblyProvider.cs [hunk 3]
```diff
@@ -42,14 +42,15 @@ public static void Initialize(string rootPath, IConditionalAssemblyContext conte
                 // todo: add a tool to update assembly versions after replacing the assemblies. (Can it support newly introduced assemblies?)
                 // todo: consider moving the list to a standalone config file
                 #region AssemblyList
-                CreateAssembly("netstandard2.0", "Azure.Core", "1.38.0.0"),
-                CreateAssembly("netstandard2.0", "Azure.Identity", "1.11.2.0"),
+                CreateAssembly("netstandard2.0", "Azure.Core", "1.40.0.0"),
+                CreateAssembly("netstandard2.0", "Azure.Core.Experimental", "0.1.0.0"),
```
Comment (isra-fel) [inline src/Accounts/AssemblyLoading/ConditionalAssemblyProvider.cs:46]: Remove Experimental once we update to the GA version of Azure.Core

File: src/Synapse/Synapse/Common/AzureSessionCredential.cs [hunk 4]
```diff
@@ -51,31 +51,36 @@ public AzureSessionCredential(IAzureContext DefaultContext, DebugLogWriter logWr
...
         public override AccessToken GetToken(TokenRequestContext requestContext, CancellationToken cancellationToken)
         {
-            AccessToken token;
+            DateTimeOffset expiresOn;
+            string token = string.Empty;
             accessToken.AuthorizeRequest((tokenType, tokenValue) =>
             {
-                token = new AccessToken(tokenValue, DateTimeOffset.UtcNow);
+                token = tokenValue;
+                expiresOn = DateTimeOffset.UtcNow;
```
Comment (isra-fel) [inline src/Synapse/Synapse/Common/AzureSessionCredential.cs:59]: I know it's not introduced by your change, but does it make sense for "expiresOn" to be "now"?

**Review by isra-fel — APPROVED** [review]
Overall it looks good to me.

File: src/Synapse/Synapse/Common/AzureSessionCredential.cs (hunk 4 above)
Comment (msJinLei) [reply src/Synapse/Synapse/Common/AzureSessionCredential.cs:59]: > I know it's not introduced by your change, but does it make sense for "expiresOn" to be "now"?

@isra-fel  we can follow up in another issue https://github.com/Azure/azure-powershell/issues/25368

**Pull request merged**